tests/main.cpp

```diff
@@ -38,37 +48,38 @@ void signalHandler(int /* signum */) {
 }
 
 uint8_t *mmap_file(char *path, int *len) {
-    int          fd;
-    int          res;
-    struct stat  sb;
-    uint8_t     *bytes;
+    int fd;
+    int res;
+    struct stat sb;
+    uint8_t *bytes;
 
     fd = open(path, O_RDONLY);
-    if (fd < 0) { FATAL("could not open file '%s'\n", path); }
+    if (fd < 0) {FATAL("could not open file '%s'\n", path); }
     res = fstat(fd, &sb);
-    if (res < 0) { FATAL("could not stat file '%s' (%d)\n", path, res); }
+    if (res < 0) {FATAL("could not stat file '%s' (%d)\n", path, res); }
 
-    bytes = (uint8_t*) mmap(0, sb.st_size, PROT_READ, MAP_SHARED, fd, 0);
+    bytes = (uint8_t *) mmap(0, sb.st_size, PROT_READ, MAP_SHARED, fd, 0);
     if (len) {
         *len = sb.st_size;  // Return length if requested
     }
-    if (bytes == MAP_FAILED) { FATAL("could not mmap file '%s'", path); }
+    if (bytes == MAP_FAILED) {FATAL("could not mmap file '%s'", path); }
     return bytes;
 }
 
 // Parse and add arguments to the stack
-void parse_args(Module *m, Type *type, int argc, char argv[][40]) {
-    for (int i=0; i<argc; i++) {
-        for (int j=0; argv[i][j]; j++) {
-            argv[i][j] = tolower(argv[i][j]); // lowecase
-        }
+void parse_args(Module *m, Type *type, int argc, Value argv[]) {
+    for (int i = 0; i < argc; i++) {
         m->sp++;
         StackValue *sv = &m->stack[m->sp];
         sv->value_type = type->params[i];
         switch (type->params[i]) {
-        case I32: sv->value.uint32 = strtoul(argv[i], NULL, 0); break;
-        case I64: sv->value.uint64 = strtoull(argv[i], NULL, 0); break;
-        case F32: if (strncmp("-nan", argv[i], 4) == 0) {
+            case I64:
+                sv->value.int64 = argv[i].int64;
+                break;
+            case I32:
+                sv->value.int32 = argv[i].int32;
+                break;
+/*        case F32: if (strncmp("-nan", argv[i], 4) == 0) {
```

Comment:
Is there something not trivial about these 2 cases?

Reply:
No there isn't. These cases have been added in b22a7ff.